### ISSUE TITLE
No such module 'Kingfisher'

### ISSUE THREAD

**Issue by KairaKrishna** [onevcat/Kingfisher#647]
### Issue Description
I am following steps https://github.com/onevcat/Kingfisher/wiki/Installation-Guide#manually. And getting 'No such module found 'Kingfisher' error.

#### What
I wish to add 'Kingfisher' as submodule under my project, and following the steps for manual installation, but getting this error. I have almost tried everything I could try, such as http://stackoverflow.com/questions/29500227/getting-error-no-such-module-using-xcode-but-the-framework-is-there; even I had re-installed latest Xcode (i.e. 8.3), but the issue still persists. I am trying to make it work since 3 days now, but no luck. 
I am attaching screenshots, just in case, those could help to figure out the error. This sample test app contains only the code visible on screen shot.

Kindly help me out with this.

Thanks.
<img width="584" alt="screen shot 2017-04-07 at 9 56 13 am" src="https://cloud.githubusercontent.com/assets/25031130/24785986/bf5375ac-1b7b-11e7-8084-32b28cfa6ef5.png">
<img width="531" alt="screen shot 2017-04-07 at 9 56 25 am" src="https://cloud.githubusercontent.com/assets/25031130/24785987/bf550b06-1b7b-11e7-8e8d-cbe83eb7452b.png">
<img width="1280" alt="screen shot 2017-04-07 at 9 57 14 am" src="https://cloud.githubusercontent.com/assets/25031130/24785988/bf599752-1b7b-11e7-8592-704c52d57b19.png">
<img width="1280" alt="screen shot 2017-04-07 at 10 20 36 am" src="https://cloud.githubusercontent.com/assets/25031130/24786014/ee1465c2-1b7b-11e7-84f8-8a91f59a575a.png">

**Comment by onevcat** [comment]
Just tried to follow the instruction and everything goes fine to me:

![snip20170407_2](https://cloud.githubusercontent.com/assets/1019875/24788022/4d6e1390-1ba6-11e7-9e80-5f6be2c1c601.png)

So I believe it is something wrong with you dev environment. Could you try to switch the scheme to Kingfisher and build it to see whether there is any error?

**Comment by KairaKrishna** [comment]
The framework is building successfully, but my sample app is not compiling. Keeps on popping the same error.
The actual problem is, in the main project which gets successfully compiled on other mac machines, but the same source code doesn't get compiled on my mac system. Always stuck at the same problem: "No such module 'Kingfisher'"

There could be any other possible system level settings/configuration/environment changes possible that could be affecting the compilation of the same source code? By same source code, I meant: I have compressed the main project and shared to other members, and exactly same code without any changes get compiled successfully.

**Comment by onevcat** [comment]
Not sure what happened. :[

**Comment by onevcat** [comment]
It is hard to say where exactly the issue is unless I could reach your environment. But it should be related to framework search path setting somewhere.

Instead of manually installing, could you give a third party dependency manager like CocoaPods or Carthage a try?

**Comment by KairaKrishna** [comment]
There are multiple dependent submodules under the main project, so I really don't wanna make a mess by changing it to pods.
Framework search path variable is set to:
$(inherited)
$(SRCROOT)

and I have checked with all possible combinations of recursive and non-recursive for both variables.

**Comment by onevcat** [comment]
:(

Sorry but have no idea now.

**Comment by shevakuilin** [comment]
I also the same situation

**Comment by lususvir** [comment]
Same here, installed with cocoapods, won't compile

**Comment by achal1012** [comment]
I also have the same issue :(

**Comment by onevcat** [comment]
@shevakuilin @lususvir @achal1012 

Thanks for reporting this. However, I have no issue in installing and using it through CocoaPods. Since Kingfisher has passed the pod lint when submitting to CocoaPods, so I believe it might be an issue related to specified development environment. Could you tell me something more about it? Like your CocoaPod and Xcode version. 

It would be great help to dig it if you can paste the result of `pod env` in the project directory here.

Thanks.

**Comment by lususvir** [comment]
I was one version behind in Xcode and Swift. So I went from 3.1 to 3.1.1 and now it compiles.

**Comment by achal1012** [comment]
I installed it using   pod 'Kingfisher', '~> 3.2.3' and it **worked**.
Here is my pod env:

### Stack

```
   CocoaPods : 1.2.0
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.4 (16E195)
       Xcode : 8.3.1 (8E1000a)
         Git : git version 2.11.0 (Apple Git-81)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 2be89b680bd9dddca8acdd0bfdd6cd1e444f295b
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.2
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
#platform :ios, '10.0'
target 'ProductCatalogue' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  # Pods for ProductCatalogue
  pod 'Firebase/Core'
  pod 'Firebase/Database'
  pod 'FSPagerView'
  pod 'Firebase/Auth'
  pod 'GoogleSignIn'
  pod 'Kingfisher', '~> 3.2.3'
  source 'https://github.com/CocoaPods/Specs.git'
  pod 'SideMenu'
end
```

**Comment by onevcat** [comment]
@lususvir @achal1012 

Thanks for confirming that. Although I am note 100% sure about it, but it seems to be an issue with unmatched Swift version and CocoaPods submitting supported version. For everyone later encountered it, I suggest to try to use the latest combination (Swift/Xcode, CocoaPods and Kingfisher) to see whether it could work or not.

Thank you again all.

**Comment by pcambre** [comment]
I had same issue, but I cleaned up my project and after re compile it every thing works well.

**Comment by jiafei1986** [comment]
I also solved this issue by clean and build the project

**Comment by c0dehunter** [comment]
I tried clean&rebuild and it didn't help the first 3 times. After shutting down XCode and restarting the computer, opening .xcworkspace fresh up and clean&build, it now works. 

Good that I am already used to XCode (iOS?) bugs, reminds me of old days when I was developing microcontroler code in KDevelop.. the worst shit I've ever been working with. And iOS development is getting close to that.

**Comment by pallavi10aggarwal** [comment]
@onevcat 
I am facing the same issue No such module 'Kingfisher' 
XCODE 10 and deployment target also 10 Swift4.2 

<img width="765" alt="screen shot 2018-09-21 at 1 43 53 pm" src="https://user-images.githubusercontent.com/37769994/45868854-6d400b00-bda4-11e8-829c-6f6ace0bf619.png">
 This is my pod file 
<img width="424" alt="screen shot 2018-09-21 at 1 44 50 pm" src="https://user-images.githubusercontent.com/37769994/45868898-8c3e9d00-bda4-11e8-81f6-960aebf7cdaf.png">

**Comment by onevcat** [comment]
@pallavi10aggarwal Could you try to upgrade your CocoaPods to 1.6.0 beta? It seems that they fixed some errors for Xcode 10, maybe it is related.

**Comment by pallavi10aggarwal** [comment]
@onevcat  I updated to beta version  but no luck , Kindly suggest some other way out .

**Comment by pallavi10aggarwal** [comment]
@onevcat  I even changed my deployment target to iOS 11 . 
No Luck , the issue still persists

**Comment by onevcat** [comment]
@pallavi10aggarwal It should be not related to your deployment target.

Is there any detail error information in your build log, except for the "No such module"? Is there any build error inside Kingfisher, or can you find the built Kingfisher.framework in the Products folder of your Pods project?

I also suggest cleaning the derived data folder then restart your Xcode to have a try.

**Comment by pallavi10aggarwal** [comment]
@onevcat  I cleaned and removed derived data several times . 
Also Along with "No such module" I  even get this error "Command CompileSwift failed with a nonzero exit code"  along with this description.
<img width="812" alt="screen shot 2018-09-25 at 10 05 30 am" src="https://user-images.githubusercontent.com/37769994/45993102-ae893100-c0aa-11e8-82cd-8b7b96ba6b95.png">

**Comment by pallavi10aggarwal** [comment]
@onevcat I cleaned my project also removed derived data.
along with  "No such module" I get this error "Command CompileSwift failed with a nonzero exit code"
with the below error description
<img width="1156" alt="screen shot 2018-09-25 at 10 21 56 am" src="https://user-images.githubusercontent.com/37769994/45993695-a54d9380-c0ad-11e8-995b-2ed3a3fa534d.png">
<img width="812" alt="screen shot 2018-09-25 at 10 05 30 am" src="https://user-images.githubusercontent.com/37769994/45993710-b8606380-c0ad-11e8-9148-635e2407fb9a.png">

**Comment by onevcat** [comment]
Could you just build your Pod project correctly? You could try it by changing your build scheme to Pods-CocoaPods or Kingfisher under the "Pods project" to see whether the framework itself gets built.

**Comment by pallavi10aggarwal** [comment]
@onevcat  the issue got resolved , I exactly don't know what solved it , a number of trials  including deleting derived data , restarting  Mac , cleaning project build  also I observed that frameworks were not there in Products folder under Pods Project , I resolved that too may be that could be the reason .
Anyways , Thanks for your support .

**Comment by onevcat** [comment]
@pallavi10aggarwal Good to hear that!

**Comment by Varuas37** [comment]
I had the similar issue. I did the following things:
1) Closed the xcode project
2) open the terminal in your project file and delete all the pods through `pod deintegrate`
3) Installed my pods again `pod install`
4) Restarted my laptop (i am not sure if that made any difference)
5) Open and build my file.

**Comment by onevcat** [comment]
@Varuas37 Maybe try to build the Pod project first to see what's the error?

**Comment by Varuas37** [comment]
@onevcat no that solved the problem for me.

**Comment by romangromov** [comment]
solved for me updating `Podfile`:
Before:
```
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'
```

After:
```
platform :ios, '10.0'
use_frameworks!
```

**Comment by romangromov** [comment]
Solved for me by doing this:

`You should open the {Project}.xcworkspace instead of the {Project}.xcodeproj after you installed anything from CocoaPods.`

This was missed from official doc, also `Kingfisher` was the first Pod I've added in my entire life.

The only question I have is: **WHY** I should open {Project}.xcworkspace instead of the {Project}.xcodeproj

**Comment by onevcat** [comment]
CocoaPods creates its own Pods.xcodeproj and builds the dependencies there. To refer to them in your app project, you need to load both your app project and the pod project. And the xcworkspace is exactly the thing to handle this: it is a container of multiple projects.

**Comment by jacksonsox** [comment]
I had this issue when creating a new project.  I was installing Kingfisher for iOS development in my pod.  What I found is that the default for my project was pointing to Mac deployment (see image).  When I changed to an iOS device, it compiled fine.

<img width="178" alt="Screen Shot 2021-03-07 at 2 50 27 PM" src="https://user-images.githubusercontent.com/32544470/110252725-c42a0280-7f54-11eb-8fe1-8581dd132f84.png">
<img width="148" alt="Screen Shot 2021-03-07 at 2 50 40 PM" src="https://user-images.githubusercontent.com/32544470/110252730-c8562000-7f54-11eb-9a7b-933dea3a5fea.png">
